ws-nextjs-app/pages/[service]/send/[id]/FormContext/index.tsx

```diff
@@ -117,26 +134,28 @@ export const FormContextProvider = ({
     const validatedData = validateFunction
       ? validateFunction(currState)
       : currState;
+    const updatedState = { [name]: { ...validatedData } };
+    const newFormState = { ...formState, ...updatedState };
 
-    setFormState(prevFormState => {
-      const updatedState = { [name]: { ...validatedData } };
-      return { ...prevFormState, ...updatedState };
-    });
+    const isFormValid = isFormStateValid(newFormState);
+
+    setHasValidationErrors(!isFormValid);
+    setFormState(newFormState);
   };
 
   const handleSubmit = async (event: FormEvent) => {
     event.preventDefault();
-    setAttemptedSubmit(true);
+    setAttemptedSubmitCount(prevCount => prevCount + 1);
```

Comment:
Does this need to be a counter? Appears to only check if its `> 0` in its uses.

Reply:
We use this counter to re-run useEffect [here](https://github.com/bbc/simorgh/blob/8fc2052904468f14a630d910db6d756fa8e5c76f/ws-nextjs-app/pages/%5Bservice%5D/send/%5Bid%5D/FormScreen/index.tsx#L58)

This take's the user's value to the Error summary box after each submission attempt. 

We found that with the boolean the focus would only go to the Error summary box on the first submission attempt. Not repeated attempts.  The counter's value changes on each submission attempt, so the useEffect hook runs on the initial and repeated submission attempts.

Reply:
Ah right, got you 👍 

I feel like we could combine some logic here to determine when the form is invalid, as we have `hasValidationErrors` and `attemptedSubmitCount` which could maybe be combined to something like `validationErrors`. 

`validationErrors` should be `[] | null` before a form is submitted, and potentially populated when it is submitted. As errors are corrected, the items in the array would be removed from `validationErrors`, so it would allow re-running of the useEffect hook to focus the error summary. I _think_ that would achieve something similar to both of these state values.

Perhaps not something for now as this looks like it works and gets us the functionality we're after. Just conscious that the state management is getting complex, but it always does for forms!

Reply:
We will likely need to store the invalidation items later anyway to display them in the summary box, but don't want to solutionise too much for the future just yet.

Reply:
Yes this makes sense. I expect this would also be helpful for populating the Error summary with a list of errors. The state is getting complex 😔. I have captured this on the ticket.